Source/Internal/IGListDisplayHandler.m

```diff
@@ -31,56 +31,95 @@ - (instancetype)init {
     return self;
 }
 
-- (void)willDisplayCell:(UICollectionViewCell *)cell
-         forListAdapter:(IGListAdapter *)listAdapter
-      sectionController:(IGListSectionController<IGListSectionType> *)sectionController
-                 object:(id)object
-              indexPath:(NSIndexPath *)indexPath {
-    IGParameterAssert(cell != nil);
+- (id)pluckObjectForView:(UICollectionReusableView *)view {
+    NSMapTable *cellObjectMap = self.visibleCellObjectMap;
+    id object = [cellObjectMap objectForKey:view];
+    [cellObjectMap removeObjectForKey:view];
+    return object;
+}
+
+- (void)willDisplayReusableView:(UICollectionReusableView *)view
+                 forListAdapter:(IGListAdapter *)listAdapter
+              sectionController:(IGListSectionController<IGListSectionType> *)sectionController
+                         object:(id)object
+                      indexPath:(NSIndexPath *)indexPath {
+    IGParameterAssert(view != nil);
     IGParameterAssert(listAdapter != nil);
     IGParameterAssert(object != nil);
     IGParameterAssert(indexPath != nil);
 
-    id <IGListDisplayDelegate> displayDelegate = [sectionController displayDelegate];
-
-    [displayDelegate listAdapter:listAdapter willDisplaySectionController:sectionController cell:cell atIndex:indexPath.item];
-
-    [self.visibleCellObjectMap setObject:object forKey:cell];
-
-    if ([self.visibleListSections countForObject:sectionController] == 0) {
-        [displayDelegate listAdapter:listAdapter willDisplaySectionController:sectionController];
+    [self.visibleCellObjectMap setObject:object forKey:view];
```

Comment:
naming again to reflect this is generic over `UICollectionReusableView` and not just cells